Description: Fix some more test cases with less-trivial failures
 Every override in this file should be accompanied by an explanation on why
 it's correct or beneficial.
Author: Ximin Luo <infinity0@debian.org>
Forwarded: TODO
---
src/sage/functions/exp_integral.py
https://www.wolframalpha.com/input/?i=root+of+Chi(z)+between+0.1+and+1.0
The overridden value is actually more correct, at least according to Wolfram
---
src/sage/rings/integer.pyx
Sage upstream expects MPIR's error output but we use GMP.
They know about this difference and advise us to just override it.
---
src/sage/numerical/optimize.py
Probably because we use slightly different versions of various num libraries.
doctest's wildcard "..." only matches against errors-greater-than but our
errors-less-than are also at a level of precision that satisfies what upstream
seems to care about (based on the number of sig-figs before the ...).
---
src/sage/coding/code_bounds.py
https://www.wolframalpha.com/input/?i=root+of+-xlog(x)%2Flog(2)+-+(1-x)log(1-x)%2Flog(2)+-+0.1+between+0+and+0.5
According to Wolfram both answers are slightly wrong so just override with ours
to reduce the noise in our test results.
--- a/sage/src/sage/functions/exp_integral.py
+++ b/sage/src/sage/functions/exp_integral.py
@@ -1170,7 +1170,7 @@
 
         sage: f(x) = cosh_integral(x)
         sage: find_root(f, 0.1, 1.0)
-        0.523822571389482...
+        0.5238225713898644...
 
     Compare ``cosh_integral(3.0)`` to the definition of the value using
     numerical integration::
--- a/sage/src/sage/rings/integer.pyx
+++ b/sage/src/sage/rings/integer.pyx
@@ -6093,8 +6093,7 @@
             sage: 1 << (2^60)
             Traceback (most recent call last):
             ...
-            MemoryError: failed to allocate ... bytes   # 64-bit
-            OverflowError: ...                          # 32-bit
+            RuntimeError: Aborted
         """
         cdef long n
 
--- a/sage/src/sage/numerical/optimize.py
+++ b/sage/src/sage/numerical/optimize.py
@@ -290,7 +290,7 @@
         sage: vars = var('x y z')
         sage: f = 100*(y-x^2)^2+(1-x)^2+100*(z-y^2)^2+(1-y)^2
         sage: minimize(f, [.1,.3,.4])
-        (1.00..., 1.00..., 1.00...)
+        (0.9999999..., 0.999999..., 0.999999...)
 
     Try the newton-conjugate gradient method; the gradient and hessian are 
     computed automatically::
@@ -329,7 +329,7 @@
         ....:    der[-1] = 200r*(x[-1r]-x[-2r]**2r)
         ....:    return der
         sage: minimize(rosen, [.1,.3,.4], gradient=rosen_der, algorithm="bfgs")
-        (1.00...,  1.00..., 1.00...)
+        (0.9999999..., 0.999999..., 0.999999...)
     """
     from sage.symbolic.expression import Expression
     from sage.ext.fast_eval import fast_callable
--- a/sage/src/sage/coding/code_bounds.py
+++ b/sage/src/sage/coding/code_bounds.py
@@ -627,7 +627,7 @@
 
         sage: from sage.coding.code_bounds import entropy_inverse
         sage: entropy_inverse(0.1)
-        0.012986862055848683
+        0.012986862055...
         sage: entropy_inverse(1)
         1/2
         sage: entropy_inverse(0, 3)
